### DISCUSSION TITLE
DGS framework and support for federation

### DISCUSSION THREAD

**Discussion by jbwtan1** [Netflix/dgs-framework#724] [Q&A] [answered]
Hi
I am planning to build a sub graph service, Netflix's dgs-framework and Apollo Federation to combine this service's schema into one federated graph schema.

The https://github.com/Netflix/dgs-framework/blob/master/README.md states that the features of the DGS framework support "Easy integration with GraphQL Federation". I can see some docs on how the DGS framework helps with testing federated queries but can anyone point me to more information on how the DGS framework helps with federation?

Does Netflix DGS framework help with integration with Apollo Federation? e.g: helping services implement "Apollo Federation schema specification" https://www.apollographql.com/docs/federation/federation-spec/#federation-schema-specification

Thank you kindly

**Comment by srinivasankavitha** [comment, marked as answer]
@jbwtan1 - The DGS framework works with Apollo's federated gateway as a service. The federation library is included already as part of the framework, and supports all the federation directives out of the box. Here is an example that shows how it would be set up to work with apollo's federated gateway: https://github.com/Netflix/dgs-federation-example

Hope that helps.
